File: Day10/Python/que1hashtable.py
def max_repeated_char(s):
    # count num of occurrences for each char
    char_count = {}
    for ch in s:
        char_count[ch] = char_count.get(ch, 0) + 1
    print(char_count)
    # find char with max occurrences
    max_count = 0
    max_char = '\0'
    for ch, count in char_count.items():
        if count > max_count:
            max_count = count
            max_char = ch
    return max_char


if __name__ == "__main__":
    s = "ABCDZBXPYDS"
    ch = max_repeated_char(s)
    print("Max Repeated Char:", ch)

File: Day10/C++/que1_hashtable.cpp
#include <iostream>
#include <vector>
#include <list>
#include <unordered_map>
using namespace std;

// Que1HashTable.cpp
char maxRepeatedChar(const string &str)
{
    // count num of occurrences for each char
    unordered_map<char, int> map;
    for (char ch : str)
    {
        map[ch]++;
    }
    // find char with max occurrences
    int maxCount = 0;
    char maxChar = '\0';
    for (const auto &pair : map)
    {
        if (pair.second > maxCount)
        {
            maxCount = pair.second;
            maxChar = pair.first;
        }
    }
    return maxChar;
}

int main()
{
    // Test Que1HashTable
    string str = "ABCDZBXPYDS";
    char ch = maxRepeatedChar(str);
    cout << "Max Repeated Char: " << ch << endl;

    return 0;
}